includes/Classifai/Watson/Linker.php

```diff
@@ -91,30 +103,27 @@ public function link_categories( $post_id, $categories ) {
 						$term = get_term_by( 'name', $part, $taxonomy );
 
 						if ( false === $term ) {
-							$term = wp_insert_term(
-								$part,
-								$taxonomy,
-								[
-									'parent' => $parent,
-								]
-							);
+							$term = wp_insert_term( $part, $taxonomy, [ 'parent' => $parent ] );
 
 							if ( ! is_wp_error( $term ) ) {
-								$parent          = intval( $term['term_id'] );
-								$terms_to_link[] = intval( $term['term_id'] );
+								$parent          = (int) $term['term_id'];
+								$terms_to_link[] = (int) $term['term_id'];
 							}
 						} else {
-							$parent          = intval( $term->term_id );
-							$terms_to_link[] = intval( $term->term_id );
+							$parent          = $term->term_id;
+							$terms_to_link[] = $term->term_id;
 						}
 					}
 				}
 			}
 		}
 
-		if ( ! empty( $terms_to_link ) ) {
+		if ( ! $return_terms ) {
```

Comment:
Seems like it would be good to keep the `! empty( $terms_to_link )` check here so we aren't trying to set an empty array of terms. If that seems right, would want to make that change in the other methods in this file